homeassistant/components/binary_sensor/concord232.py

```diff
@@ -0,0 +1,148 @@
+"""
+Support for exposing Concord232 elements as sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.concord232/
+"""
+import datetime
+
+import logging
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA, SENSOR_CLASSES)
+from homeassistant.const import (CONF_HOST, CONF_PORT)
+
+import homeassistant.helpers.config_validation as cv
+
+import requests
+
+import voluptuous as vol
+
+
+REQUIREMENTS = ['concord232==0.14']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_EXCLUDE_ZONES = 'exclude_zones'
+CONF_ZONE_TYPES = 'zone_types'
+
+DEFAULT_HOST = 'localhost'
+DEFAULT_PORT = '5007'
+DEFAULT_SSL = False
+
+ZONE_TYPES_SCHEMA = vol.Schema({
+    cv.positive_int: vol.In(SENSOR_CLASSES),
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_EXCLUDE_ZONES, default=[]):
+        vol.All(cv.ensure_list, [cv.positive_int]),
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_ZONE_TYPES, default={}): ZONE_TYPES_SCHEMA,
+})
+
+CONCORD232_GLOBAL = None
+
+SCAN_INTERVAL = 1
+
+DEFAULT_NAME = "Alarm"
+
+
+# pylint: disable=too-many-locals
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Concord232 binary sensor platform."""
+    from concord232 import client as concord232_client
+
+    def get_opening_type(zone):
+        """Helper function to try to guess sensor type frm name."""
+        if "MOTION" in zone["name"]:
+            return "motion"
+        if "KEY" in zone["name"]:
+            return "safety"
+        if "SMOKE" in zone["name"]:
+            return "smoke"
+        if "WATER" in zone["name"]:
+            return "water"
+        return "opening"
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    exclude = config.get(CONF_EXCLUDE_ZONES)
+    zone_types = config.get(CONF_ZONE_TYPES)
+    sensors = []
+
+    try:
+        global CONCORD232_GLOBAL
```

Comment:
I'm declaring a global as both the alarm panel and the binary sensors use it,  thus to minimize calls to the API from each individual sensor, and the alarm panel itself, i created the global to do that work.  

Does this approach not do that?